pkg/cmd/experimental/syncgroups/groupsyncer.go

```diff
@@ -0,0 +1,260 @@
+package syncgroups
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-ldap/ldap"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+
+	"github.com/openshift/origin/pkg/client"
+	syncconfig "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/schema1"
+	ouserapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// GroupSyncer runs a Sync job on Groups
+type GroupSyncer interface {
+	Sync() (err error)
```

Comment:
I haven't read all the way down, but this definitely feels like a case where we'd want `[]error`